incubator/concourse/templates/worker-statefulset.yaml

```diff
@@ -0,0 +1,116 @@
+                      "key": "app",
+                      "operator": "In",
+                      "values": ["{{ template "worker.fullname" . }}"]
+                    }]
+                  },
+                  "topologyKey": "kubernetes.io/hostname"
+                }
+              ]
+            }
+          }
+    spec:
+      terminationGracePeriodSeconds: 60
+      containers:
+        - name: {{ template "worker.fullname" . }}
+          image: "{{ .Values.image }}:{{ .Values.imageTag }}"
+          imagePullPolicy: {{ default "" .Values.imagePullPolicy | quote }}
+          args:
+            - "worker"
+          lifecycle:
+            preStop:
+              exec:
+                command:
+                  - "concourse"
+                  - "land-worker"
+                  - "--name=${HOSTNAME}"
+          env:
+            - name: CONCOURSE_TSA_HOST
+              valueFrom:
+                configMapKeyRef:
+                  name: {{ template "concourse.fullname" . }}
+                  key: concourse-tsa-host
+            - name: CONCOURSE_TSA_PORT
+              valueFrom:
+                configMapKeyRef:
+                  name: {{ template "concourse.fullname" . }}
+                  key: concourse-tsa-port
+            - name: CONCOURSE_GARDEN_DOCKER_REGISTRY
+              valueFrom:
+                configMapKeyRef:
+                  name: {{ template "concourse.fullname" . }}
+                  key: garden-docker-registry
+            - name: CONCOURSE_GARDEN_INSECURE_DOCKER_REGISTRY
+              valueFrom:
+                configMapKeyRef:
+                  name: {{ template "concourse.fullname" . }}
+                  key: garden-insecure-docker-registry
+            - name: CONCOURSE_TSA_PUBLIC_KEY
+              value: "/concourse-keys/host_key.pub"
+            - name: CONCOURSE_TSA_WORKER_PRIVATE_KEY
+              value: "/concourse-keys/worker_key"
+            - name: CONCOURSE_WORK_DIR
+              value: "/concourse-work-dir"
+          resources:
+{{ toYaml .Values.worker.resources | indent 12 }}
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - name: concourse-keys
+              mountPath: /concourse-keys
+              readOnly: true
+            - name: concourse-work-dir
+              mountPath: /concourse-work-dir
+      volumes:
+        - name: concourse-keys
+          secret:
+            secretName: {{ template "concourse.fullname" . }}
+            defaultMode: 0400
+            items:
+              - key: host-key-pub
+                path: host_key.pub
+              - key: worker-key
+                path: worker_key
+              - key: worker-key-pub
+                path: worker_key.pub
+  {{- if .Values.persistence.enabled }}
+  volumeClaimTemplates:
+    - metadata:
+        name: concourse-work-dir
+        annotations:
+          volume.alpha.kubernetes.io/storage-class: {{ .Values.persistence.worker.class | quote }}
```

Comment:
The `volume.alpha` version of the `storage-class` annotation doesn't actually use its value. To support both the new behavior (which requires a value) and the old behavior, we recommend a pattern like this:

https://github.com/kubernetes/charts/pull/361/commits/95626d44db84935804aa029a573c8c75be2090f3